test/e2e/realtimecompositions_test.go

```diff
@@ -0,0 +1,111 @@
+		Kind:       "NopResource",
+	}))
+	nopList := composed.NewList(composed.FromReferenceToList(corev1.ObjectReference{
+		APIVersion: "realtime-revision-selection.e2e.crossplane.io/v1alpha1",
+		Kind:       "NopResource",
+	}))
+	xnopList := composed.NewList(composed.FromReferenceToList(corev1.ObjectReference{
+		APIVersion: "realtime-revision-selection.e2e.crossplane.io/v1alpha1",
+		Kind:       "XNopResource",
+	}))
+	withTestLabels := resources.WithLabelSelector(labels.FormatLabels(map[string]string{"realtime-compositions": "true"}))
+
+	environment.Test(t,
+		features.New(t.Name()).
+			WithLabel(LabelStage, LabelStageAlpha).
+			WithLabel(LabelArea, LabelAreaAPIExtensions).
+			WithLabel(LabelSize, LabelSizeSmall).
+			WithLabel(LabelModifyCrossplaneInstallation, LabelModifyCrossplaneInstallationTrue).
+			WithLabel(config.LabelTestSuite, SuiteRealtimeCompositions).
+			WithSetup("EnableAlphaRealtimeCompositions", funcs.AllOf(
+				funcs.AsFeaturesFunc(environment.HelmUpgradeCrossplaneToSuite(SuiteRealtimeCompositions)),
+				funcs.ReadyToTestWithin(1*time.Minute, namespace),
+			)).
+			WithSetup("PrerequisitesAreCreated", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "setup/*.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "setup/*.yaml"),
+				funcs.ResourcesHaveConditionWithin(1*time.Minute, manifests, "setup/definition.yaml", apiextensionsv1.WatchingComposite()),
+				funcs.ResourcesHaveConditionWithin(2*time.Minute, manifests, "setup/provider.yaml", pkgv1.Healthy(), pkgv1.Active()),
+			)).
+			Assess("CreateClaim", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "claim.yaml"),
+				funcs.InBackground(funcs.LogResources(nopList, withTestLabels)),
+				funcs.InBackground(funcs.LogResources(xnopList, withTestLabels)),
+				funcs.InBackground(funcs.LogResources(nopCrossplaneList, withTestLabels)),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "claim.yaml"),
+				funcs.ResourcesHaveConditionWithin(5*time.Minute, manifests, "claim.yaml", xpv1.Available()),
+			)).
+			Assess("UpdateMR", funcs.AllOf(
+				funcs.ListedResourcesModifiedWith(nopCrossplaneList, 1, func(object k8s.Object) {
+					anns := object.GetAnnotations()
+					if anns == nil {
+						anns = make(map[string]string)
+					}
+					anns["cool-field"] = "I'M COOL!"
+					object.SetAnnotations(anns)
+				}, withTestLabels),
+			)).
+			Assess("ClaimHasPatchedField",
+				// 10 seconds is a long time for a realtime composition, but
+				// considerably below the normal reconcile time for a XR.
+				funcs.ResourcesHaveFieldValueWithin(10*time.Second, manifests, "claim.yaml", "status.coolerField", "I'M COOL!"),
+			).
+			WithTeardown("DeletePrerequisites", funcs.AllOf(
+				funcs.DeleteResources(manifests, "setup/*.yaml"),
+				funcs.ResourcesDeletedWithin(3*time.Minute, manifests, "setup/*.yaml"),
+			)).
+			// Disable our feature flag.
+			WithTeardown("DisableAlphaUsages", funcs.AllOf(
```

Comment:
you are not disabling usages feature here, right?

Reply:
oops, great spot.

Reply:
No, actually this was just cosmetics (feature name). Had no impact.

Reply:
done